sw/host/tests/rom/e2e_chip_specific_startup/src/main.rs

```diff
@@ -26,6 +31,270 @@ struct Opts {
         help = "Console receive timeout",
     )]
     timeout: Duration,
+
+    #[structopt(long, help = "OTP is unprogrammed; be permissive with OTP values")]
+    otp_unprogrammed: bool,
+}
+
+fn check_ast(opts: &Opts, cs: &ChipStartup) -> Result<()> {
+    let en = MultiBitBool4::try_from(cs.otp.creator_sw_cfg_ast_init_en)?;
+    log::info!("CREATOR_SW_CFG_AST_INIT_EN = {:#x}", en);
+    // AST_INIT_EN should be either MultiBitBool4 True or False.  Other values are an error.
+    match en {
+        MultiBitBool4::True => assert_eq!(cs.ast_init_done, true),
+        MultiBitBool4::False => assert_eq!(cs.ast_init_done, false),
+        _ => {
+            if opts.otp_unprogrammed {
+                log::info!(
+                    "CREATOR_SW_CFG_AST_INIT_EN is neither True nor False:  AST not checked."
+                );
+            } else {
+                assert!(en.is_known_value());
+            }
+        }
+    }
+    Ok(())
+}
+
+fn check_jitter(opts: &Opts, cs: &ChipStartup) -> Result<()> {
+    let en = MultiBitBool4::try_from(cs.otp.creator_sw_cfg_jitter_en)?;
+    log::info!("CREATOR_SW_CFG_JITTER_EN = {:#x}", en);
+    // Clock jitter should be disabled if JITTER_EN is MultiBitBool4 False. All other values
+    // enable.
+    match en {
+        MultiBitBool4::False => assert_eq!(cs.jitter, false),
+        MultiBitBool4::True => assert_eq!(cs.jitter, true),
+        _ => {
+            if opts.otp_unprogrammed {
+                log::info!("CREATOR_SW_CFG_AST_JITTER_EN is neither True nor False:  Skipping jitter check.");
+                // TODO: The documentation says that jitter should be true for any unknown values.
```

Comment:
I think jitter can still be enabled for all other values but the register probably shows the actual value that was written.

Reply:
There is a bug in the DIF then: the function `dif_clkmgr_jitter_get_enabled` was using `dif_multi_bit_bool_to_toggle` to convert the mubi value to a `dif_toggle_t`.  That function, however, only accepts mubi `True` values as enabled and regards all other values as disabled.

Fixed the DIF.

Reply:
SG